File: system.go
package orbo

import (
	"context"
	"log"
	system "orbo/gen/system"
)

// system service example implementation.
// The example methods log the requests and return zero values.
type systemsrvc struct {
	logger *log.Logger
}

// NewSystem returns the system service implementation.
func NewSystem(logger *log.Logger) system.Service {
	return &systemsrvc{logger}
}

// Get overall system status
func (s *systemsrvc) Status(ctx context.Context) (res *system.SystemStatus, err error) {
	res = &system.SystemStatus{}
	s.logger.Print("system.status")
	return
}

// Start motion detection on all active cameras
func (s *systemsrvc) StartDetection(ctx context.Context) (res *system.SystemStatus, err error) {
	res = &system.SystemStatus{}
	s.logger.Print("system.start_detection")
	return
}

// Stop motion detection
func (s *systemsrvc) StopDetection(ctx context.Context) (res *system.SystemStatus, err error) {
	res = &system.SystemStatus{}
	s.logger.Print("system.stop_detection")
	return
}

File: web/frontend/src/api/system.ts
import { get, post } from './client';
import type { SystemStatus } from '../types';

export async function getSystemStatus(): Promise<SystemStatus> {
  return get<SystemStatus>('/system/status');
}

export async function startDetection(): Promise<void> {
  return post<void>('/system/detection/start');
}

export async function stopDetection(): Promise<void> {
  return post<void>('/system/detection/stop');
}
